diff --git a/src/fr-window.c b/src/fr-window.c
index 7aa574f..c08fffd 100644
--- a/src/fr-window.c
+++ b/src/fr-window.c
@@ -4925,7 +4925,7 @@ fr_window_construct (FrWindow *window)
 	adw_button_content_set_label (ADW_BUTTON_CONTENT (content), C_("Action", "_Extract…"));
 	adw_button_content_set_icon_name (ADW_BUTTON_CONTENT (content), "archive-extract-symbolic");
 	adw_button_content_set_use_underline (ADW_BUTTON_CONTENT (content), true);
-	gtk_actionable_set_action_name (GTK_ACTIONABLE (button), "win.extract-all-by-default");
+	gtk_actionable_set_action_name (GTK_ACTIONABLE (button), "win.extract-files");
 	gtk_button_set_child (GTK_BUTTON (button), content);
 	gtk_widget_set_tooltip_text (button, _("Extract Files"));
 	gtk_size_group_add_widget (header_bar_size_group, button);
